main/models.py

```diff
@@ -1,40 +1,149 @@
 from django.db import models
 from django.utils import timezone
-from django.contrib.auth.models import AbstractUser
+from django.contrib.auth.models import (AbstractUser,BaseUserManager)
 from django.utils.translation import gettext_lazy as _
+import uuid
+import logging
+import gocardless_pro
+
+# get instance of a logger
+logger = logging.getLogger(__name__)
+
+
+class SpaceUserManager(BaseUserManager):
+    def create_user(self, email, password=None, space=None):
+        """
+        Creates and saves a User with the given email, space
+        and password.
+        """
+        if not email:
+            raise ValueError('Users must have an email address')
+
+        user = self.model(
+            email=self.normalize_email(email),
+            space=space,
+        )
+
+        if password:
+            user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_superuser(self, email, password):
+        """
+        Creates and saves a superuser with the given email
+        and password.
+        """
+        user = self.create_user(
+            email,
+            password=password,
+        )
+        user.is_admin = True
+        user.is_staff = True
+        user.is_superuser = True
+        user.save(using=self._db)
+        return user
 
 class User(AbstractUser):
 
-    SPACE_STATUS_CHOICES = (
+    APPROVAL_STATUS_CHOICES = (
         ("Blank", "Blank"),   #space is blank
         ("Pending", "Pending"),  # space relationship has changed, approval is pending
         ("Approved", "Approved"),  # space relationship has been approved
         ("Rejected", "Rejected"),  # space relationship has been rejected
     )
 
+    MEMBER_TYPE_CHOICES = (
+        ("None", "None"),  # i.e. a manager of profile info
+        ("Supporter", "Supporter"),
+        ("Representative", "Representative"),
+    )
+
     # disable default username field
     username = None
     # add email, make it the unique field
     email = models.EmailField(_('email address'), unique=True)
+    # override default
+    USERNAME_FIELD = 'email'
+
+    # whether user account is active (i.e. password has been reset after initial signup)
+    active = models.BooleanField(default=False)
+
+    # what type of member is this?
+    member_type = models.CharField(max_length=14, choices=MEMBER_TYPE_CHOICES, default='None')
+    # member application status
+    member_status = models.CharField(max_length=8, choices=APPROVAL_STATUS_CHOICES, default='Blank')
```

Comment:
The info for individual memberships should probably live in an IndividualMembership table rather than the user table, as not every user will have them. (We could probably also have a SpaceMembership table to hold that state for spaces.)